validation/src/test/scala/hmda/validation/rules/lar/macro/Q007Spec.scala

```diff
@@ -5,28 +5,38 @@ import com.typesafe.config.ConfigFactory
 import hmda.model.fi.lar.LoanApplicationRegister
 import hmda.validation.rules.AggregateEditCheck
 import hmda.validation.rules.lar.`macro`.MacroEditTypes.LoanApplicationRegisterSource
-import org.scalacheck.Gen
 
 class Q007Spec extends MacroSpec {
 
   val config = ConfigFactory.load()
-  val multiplier = config.getDouble("hmda.validation.macro.Q007.numOfLarsMultiplier") * 100
+  val multiplier = config.getDouble("hmda.validation.macro.Q007.numOfLarsMultiplier")
 
-  val multiplierGen: Gen[Int] = Gen.choose(1, multiplier.toInt)
+  val testLars = lar100ListGen.sample.getOrElse(Nil)
+  val sampleSize = testLars.size
 
-  property("be valid if not accepted <= 0.15 * total") {
-    larSource.mustPass
+  property(s"be valid if not accepted < $multiplier * total") {
+    val numOfGoodLars = (sampleSize * multiplier).toInt - 1
+    val newLarSource = newSource(numOfGoodLars)
+    newLarSource.mustPass
   }
 
-  property("be invalid if withdrawn > 0.15 * total") {
-    forAll(multiplierGen) { multiplier =>
-      val badLar = lars.head.copy(actionTakenType = 2)
-      val badLars = Array.fill(multiplier)(badLar)
-      val goodLars = Array.fill(100 - multiplier)(lars.head)
-      val newLars = badLars ++ goodLars
-      val newLarSource = Source.fromIterator(() => newLars.toIterator)
-      newLarSource.mustFail
-    }
+  property(s"be valid if not accepted = $multiplier * total") {
+    val numOfGoodLars = (sampleSize * multiplier).toInt
+    val newLarSource = newSource(numOfGoodLars)
+    newLarSource.mustPass
+  }
+
+  property(s"be invalid if not accepted > $multiplier * total") {
+    val numOfGoodLars = (sampleSize * multiplier).toInt + 1
+    val newLarSource = newSource(numOfGoodLars)
+    newLarSource.mustFail
+  }
+
+  private def newSource(numOfGoodLars: Int) = {
```

Comment:
This specific implementation is only relevant to a certain edit based on `actionTakenType`.  There might be a way to pull out common code, but I wouldn't block the PR based on that.